ledger/catchupaccessor.go

```diff
@@ -298,11 +298,40 @@ func (c *catchpointCatchupAccessorImpl) ProcessStagingBalances(ctx context.Conte
 	if strings.HasPrefix(sectionName, "balances.") && strings.HasSuffix(sectionName, ".msgpack") {
 		return c.processStagingBalances(ctx, bytes, progress)
 	}
+	if sectionName == "stateProofVerificationData.msgpack" {
+		return c.processStagingStateProofVerificationData(ctx, bytes, progress)
+	}
 	// we want to allow undefined sections to support backward compatibility.
 	c.log.Warnf("CatchpointCatchupAccessorImpl::ProcessStagingBalances encountered unexpected section name '%s' of length %d, which would be ignored", sectionName, len(bytes))
 	return nil
 }
 
+// processStagingStateProofVerificationData deserialize the given bytes as a temporary staging state proof verification data
+func (c *catchpointCatchupAccessorImpl) processStagingStateProofVerificationData(_ context.Context, bytes []byte, _ *CatchpointCatchupAccessorProgress) (err error) {
+	var decodedData catchpointStateProofVerificationData
+	err = protocol.Decode(bytes, &decodedData)
+	if err != nil {
+		return err
+	}
+
+	wdb := c.ledger.trackerDB().Wdb
+
+	// 6 months of stuck state proofs should lead to about 1.2 MB of data, so we avoid redundant timers
+	// and progress reports.
+
+	err = wdb.Atomic(func(ctx context.Context, tx *sql.Tx) (err error) {
+		for _, data := range decodedData.Data {
+			err = writeCatchpointStateProofVerificationData(ctx, tx, &data)
+			if err != nil {
+				return err
+			}
+		}
+		return
+	})
+	// TODO: Set synchronous mode in db?
+	return err
```

Comment:
Should I change the synchronous mode?

Reply:
I'd prefer to leave it as is, so as to not have to implement different logic for old catchpoints, for example.